File: spring-boot-backend/src/main/java/com/eafc/springbootbackend/entities/product/Discount.java
package com.eafc.springbootbackend.entities.product;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Collection;

@Entity
@Data
public class Discount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @NotNull
    @NotBlank
    private String discountName;
    @NotNull
    private int percentage;
    private boolean isActive;

    @OneToMany(mappedBy = "discount")
    private Collection<ProductInfo> discountedProducts;
}

File: angular-frontend/src/app/common/prod-details/discount.ts
import {Stock} from "./stock";
import {SubCategory} from "./sub-category";
import {ProductInfo} from "../shopping/product-info";

export class Discount {
  public discountId!: number;
  public discountName!: string;
  public percentage!: number;
  public isActive!: boolean;
  public discountedProducts!: ProductInfo[];
}
